### PR TITLE
docker file for centos

### PR DESCRIPTION
**Pull request by swilly22**

### CODE REVIEW

File: Dockerfile.centos [hunk 1]
```diff
@@ -0,0 +1,48 @@
+FROM centos:7 AS builder
+
+# Install prerequisite
+RUN yum install -y build-base musl-dev automake make cmake autoconf libtool wget git m4 libgomp
```
Comment (rafie) [inline Dockerfile.centos:4]: Why musl-dev?

File: Dockerfile.centos [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    cd peg-0.1.18;\
+    make; make install
+
+# Clone RedisGraph
+RUN cd /
+RUN git clone --recurse-submodules https://github.com/RedisGraph/RedisGraph.git
+
+WORKDIR /RedisGraph
+
+# Upgrade GCC & Build RedisGraph
```
Comment (rafie) [inline Dockerfile.centos:18]: Better install new GCC before building PEG.

File: Dockerfile.centos [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+RUN yum install -y centos-release-scl devtoolset-7 scl-utils
+RUN yum install -y devtoolset-7
```
Comment (rafie) [inline Dockerfile.centos:20]: There's a newer devtoolset 8.

File: Dockerfile.centos [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+RUN source scl_source enable devtoolset-7;\
+    make clean; \
+    make
+
+# Build redis
+WORKDIR /
+RUN wget http://download.redis.io/releases/redis-5.0.8.tar.gz;\
```
Comment (rafie) [inline Dockerfile.centos:27]: 6.0.3

File: Dockerfile.centos [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    tar xzf redis-5.0.8.tar.gz;\
+    cd redis-5.0.8;\
+    make
+
+FROM centos:7
```
Comment (rafie) [inline Dockerfile.centos:32]: IMHO it's better to start from `redisfab/redis:6.0.1-x64-centos7`.

**Comment by swilly22** [comment]
Thank you @rafie! I'll merge for now and address to your comments later on.

**Pull request merged**